dplace_app/api_views.py

```diff
@@ -320,13 +333,22 @@ def id_array(l):
 
     log.info('mid 1: %s' % (time() - _s,))
 
-    # search for language trees
-    labels = models.LanguageTreeLabels.objects.filter(societies__id__in=soc_ids).all()
-    log.info('mid 3: %s' % (time() - _s,))
-
-    global_tree = None
-    global_newick = []
-    global_isolates = []
+    # Filter the results to those that matched all criteria
+    #result_set.finalize(criteria)
+    log.info('mid 2: %s' % (time() - _s,))
+    return result_set
+    
+@api_view(['GET'])
+@permission_classes((AllowAny,))
+def trees_from_societies(request):
+    language_trees = []
+    for k, v in request.query_params.lists():
```

Comment:
This `for`-loop is supposed to only run once, right? Could this be clarified somehow? E.g. by checking on `k`?